### ISSUE TITLE
Datagrid: Checkboxes inside component formatters are not clickable

### ISSUE THREAD

**Issue by bartlomiej-k** [infor-design/enterprise-ng#1597]
**Describe the bug**
When a component used as a formatter or editor component of a column contains a soho-checkbox, this checkbox cannot be checked/unchecked from the UI. From user perspective, it looks like the cell intercepts the click and doesn't propagate it to the checkbox, even though the cursor changes its style to pointer only when hovering over the checkbox.

Our use case is to display up to 10 columns containing checkboxes that act as true/false flags of edited objects, so trying to configure checkbox columns like they were 'select row' checkboxes is not a solution for us.

**To Reproduce**
1. Go to https://stackblitz.com/edit/ids-quick-start-1701-7vtkax
2. Click on any cell in _Editor checkbox_ column to open editor component.
3. Click checkbox.
4. Observe checkbox doesn't change its value.
5. Click any other cell or press enter to get out of the edit cell mode.
6. Click on any checkbox in _Formatter checkbox_ column.
7. Observe checkbox doesn't change its value.

**Expected behavior**
Checkbox changes its value, just like it does in the same component placed below the datagrid.

**Version**
- ids-enterprise-ng: v17.0.1

**Platform**
 - Infor Application/Team Name: Infor WFM

**Comment by tmcconechy** [comment]
This is probably going to be really hard to solve using an angular component for this. Why not use the one that already works?

https://main-enterprise.demo.design.infor.com/components/datagrid/test-editable-checkboxes.html (can you point at the difference?)

I.E. built in editor and formatter for checkbox not an angular one instead..

**Comment by bartlomiej-k** [comment]
One checkbox per cell, but up to 10 columns can contain a checkbox.

Our component has many more functionalities than built-in function formatter, like security settings per user, default values, additional validation. Creating custom formatter function to handle all of that is basically recreating the whole component into a function and with all the dependencies it has, it's just impossible to do.

For sure, in theory we _could_ create a library/service that would translate all our custom components into functions, so the datagrid handles them correctly (what would take months during which both jQuery and Angular libriaries will lose support), but it leads to a question - what's the purpose of formatter/editor component column setting if whenever we try to use it, it causes major issues even though we use IDS components for every interactable element on the screen.

I wonder whether this special click handling could be skipped in some way for columns containing components. If I remember correctly, it already contains some code for special treatment of cells marked as _inlineEditor_ which could be used as a workaround if only the target of the click event was input element, not the associated label element.

**Comment by tmcconechy** [comment]
Yeah the special angular editor formatter was just thrown in as an external contribution. Its always not worked very well. Maybe we can rethink this in the new web components and make it a more first class feature.

The problem is the grid dynamically creates this cell a lot and its slow too.

Maybe we can make this example work as described by hacking it up. Honestly i'd try to work the logic into existing string style editors and formatters if possible. But we can take a look as time will allow. This wont be a simple fix i think in the current enterprise components but marked it for review as such.

**Comment by bartlomiej-k** [comment]
Knowing how the datagrid handles the components and how many bugs and inconsistencies there are, I understand where all of this comes from and how difficult it would be to fix basically any of these issues. 

Implementing such mechanism into the web component version would also be a nightmare as it's supposed to support more languages and frameworks than pure javascript and Angular. And it's something we want to request as well.

What would also do the work for us is having the datagrid not as a dynamic component with the whole API, but as a simple table with the styling of a datagrid - no javascript, just HTML structure and CSS styling shared with the original datagrid, so when there is a styling update, the _static_ version of the datagrid would automatically have its styling updated. That way we could use our components the same way we use them outside of the datagrid - attaching custom event listeners, validation functions, filling up with any values we want, wrapping them in a form. No need to call any datagrid functions to update dataset, sort rows, reorder columns etc. as we can do it on our own. Like it was a fieldset, but displayed as a table with datagrid styling.

This is how it could be used if it was implemented as a web component in the simplest way:
```
<!-- Define base grid - id, row height, maybe auto-fit if it's CSS only solution -->
<ids-data-grid-static id="grid-static" row-height="md" list-style="true">

  <!-- Headers definitions -->
    <ids-data-grid-static-column-header width="150px">
        <ids-text>Some header</ids-text>
    </ids-data-grid-static-column-header>
    <ids-data-grid-static-column-header width="250px">
        <ids-text>Some other header</ids-text>
    </ids-data-grid-static-column-header>

    <!-- Loop through rows in any language you want - in our case, it's Angular loop -->
    <ng-container *ngFor="let row of rows">
    <!-- Define as many columns as you need to show -->
        <ids-data-grid-static-column class="some-custom-class" width="150px" frozen="true">
            <some-component [value]="row.value"></some-component>
        </ids-data-grid-static-column>
        <ids-data-grid-static-column width="250px">
            <other-component [displayValue]="row.attributeToDisplay"></other-component>
        </ids-data-grid-static-column>
    </ng-container>

</ids-data-grid-static>
```

We already have a custom component that can do all these things, but we don't want to use datagrid classes to style it as we already had many styling issues after library upgrades. Having that as a part of the library makes it much easier to maintain.

As for hacking the datagrid, I already spent already two months trying to hack it to make all our components work inside the datagrid and there are still many things that simply don't work. Even though most of them are some CSS issues, there are still some features requested by our BAs that require even more hacking. It would be a relief to have a simple component that would act as a simple layout provider and nothing more.

**Comment by tmcconechy** [comment]
The problem with that might be all the functionality like selection and filtering and make it all working. Unless you want a completely readonly datagrid? That doesnt seem much easier but its worth considering adding more to our css only solution https://main.wc.design.infor.com/ids-data-grid/standalone-css.html or making it more composable somehow.. Maybe with an ngFor some limited things still work?

As far as hacking it up to make the one use case work we can try that. But i think other use cases will be a problem. BTW for this case as written. Do you need the checkbox to show / how do they know they can click it or is it only as entered? 

Maybe we need to explore the use cases and see why they are not possible and balance them with not doing them/adding something to make them easier. So far the string and editor we made do and can work. So maybe just need to request them more one by one and see if we can get them to work?

**Comment by bartlomiej-k** [comment]
We don't need any functionalities of the grid, styling is enough. 

I had a look at this example with standalone grid css and this is exactly what we need, but in a bit different way. Instead of manually recreating the grid styling, I see it as a component that just imports the styling of the grid and resolves itself to a plain HTML tag with a given formatting. So `<ids-data-grid-static>` is replaced by `<table class="ids-data-grid">`, `<ids-data-grid-header-static>` by `<tr class="ids-data-grid-header">` etc. The advantages of that would be (1) that we don't need to manually append styling of the datagrid to the DOM like this example does, because it's already encapsulated in the component, (2) we don't need to know all dependencies and stylesheets of the datagrid, it handles it and applies the styling on its own and (3) in case of any styling update (new class, renamed existing class, new defaults etc.), we don't need to update all our usages of the datagrid as it will be done automatically by the component (in other words, by you, but in one place 😉). That would be a completely readonly datagrid that contains styling (maybe even partially configurable, like row height) and nothing else.

Our checkboxes don't need to show they are clickable in any way - users will know they can click a checkbox based on its enabled/disabled property. We can handle that internally in the component. But yeah, with all the logic for formatter/editor components, row selection, expansion etc., just triggering the click for the component below might get tricky. Or maybe that would be the way - propagate the click to the component and let it handle the event, without searching for any focusable elements...

**Comment by tmcconechy** [comment]
We can consider it. I think its a lot trickier than you think. And also 2. would never be possible. You would still need to import the datagrid.

I think overall its worth considering but no promises.
